relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,101 @@
+use std::sync::Arc;
+use std::time::Duration;
+
+use relay_dynamic_config::GlobalConfig;
+use relay_system::{Addr, AsyncResponse, FromMessage, Interface, Sender, Service};
+
+use crate::actors::processor::EnvelopeProcessor;
+use crate::actors::project_upstream::GetProjectStates;
+use crate::actors::upstream::{SendQuery, UpstreamRelay};
+
+/// Service implementing the [`GlobalConfigResponse`] interface.
+///
+/// The service is responsible for fetching the global config and
+/// forwarding it to the services that require it.
+#[derive(Debug)]
+pub struct GlobalConfigService {
+    global_config: Arc<GlobalConfig>,
+    project_cache: Addr<EnvelopeProcessor>,
+    upstream: Addr<UpstreamRelay>,
+}
+
+/// Service interface for the [`GetGlobalConfig`] message.
+pub struct GlobalConfigResponse(Sender<Arc<GlobalConfig>>);
+
+pub struct GetGlobalConfig;
+
+impl FromMessage<GetGlobalConfig> for GlobalConfigResponse {
+    type Response = AsyncResponse<Arc<GlobalConfig>>;
+
+    fn from_message(_message: GetGlobalConfig, sender: Sender<Arc<GlobalConfig>>) -> Self {
+        Self(sender)
+    }
+}
+
+impl Interface for GlobalConfigResponse {}
+
+impl GlobalConfigService {
+    pub fn new(project_cache: Addr<EnvelopeProcessor>, upstream: Addr<UpstreamRelay>) -> Self {
+        let global_config = Arc::new(GlobalConfig::default());
+        Self {
+            global_config,
+            project_cache,
+            upstream,
+        }
+    }
+
+    /// Forwards the given global config to the services that require it.
+    async fn update_global_config(&mut self) {
+        let upstream_relay: Addr<UpstreamRelay> = self.upstream.clone();
+        let project_cache = self.project_cache.clone();
+
+        let query = GetProjectStates {
+            public_keys: vec![],
+            full_config: false,
+            no_cache: false,
+            global_config: true,
+        };
```

Comment:
now we are using the same query both for project cache and global config to the sentry endpoint